src/roiextractors/multiimagingextractor.py

```diff
@@ -0,0 +1,107 @@
+from abc import ABC
+from array import ArrayType
+from typing import Tuple, List, Iterable
+
+import numpy as np
+
+from .extraction_tools import NumpyArray
+from .imagingextractor import ImagingExtractor
+
+
+class MultiImagingExtractor(ImagingExtractor, ABC):
+    """
+    This class is used to combine multiple ImagingExtractor objects by frame.
+    """
+
+    extractor_name = "MultiImagingExtractor"
+    installed = True
+    installation_mesg = ""
+
+    def __init__(self, imaging_extractors: List[ImagingExtractor]):
+        """
+        Parameters
+        ----------
+        imaging_extractors: list of ImagingExtractor
+            list of imaging extractor objects
+        """
+        super().__init__()
+        assert isinstance(imaging_extractors, list), "Enter a list of ImagingExtractor objects as argument"
+        assert all(isinstance(IX, ImagingExtractor) for IX in imaging_extractors)
+        self._imaging_extractors = imaging_extractors
+
+        # Checks that properties are consistent between extractors
+        self._check_consistency_between_imaging_extractors()
+
+        # Set properties based off the initial extractor
+        self._first_imaging_extractor = self._imaging_extractors[0]
+        self._num_channels = self._first_imaging_extractor.get_num_channels()
+        self._channel_names = self._first_imaging_extractor.get_channel_names()
+        self._sampling_frequency = self._first_imaging_extractor.get_sampling_frequency()
+        self._image_size = self._first_imaging_extractor.get_image_size()
```

Comment:
I don't mind much either way, but my general approach to this kind of thing is to reduce the amount of internal allocation in the way of class attributes and instead farm out the API calls respectively.

What I mean by that in this context is, instead of pre-extracting a private attribute such as `self._num_channels` and then having `def get_num_channels` simply return that private attribute (*i.e.*, basically just setting these as [class properties](https://realpython.com/python-property/)), I would personally not set any properties and instead define such methods like

```python
def get_num_channels:
    return self._imaging_extractors[0].get_num_channels()
```

Note this is perfectly safe to do since any instance of this class has passed `self._check_consistency_between_imaging_extractors()`.

This has some advantages such as; if scanning the attributes of an instance of this class then there's fewer items to search through, and it reduces the lines of code needed as well. It also makes the size of an instance of this class smaller, though that's less relevant since these are all near-scalar quantities.

Again, it's not a big deal either way so if you prefer the attribute approach feel free to keep it.

Reply:
I see your point and agree, I'll make the changes. Thank you for all the suggestions!